hack/run_test

```diff
@@ -480,21 +498,30 @@ $volume_options
   echo "  Success!"
 }
 
+function run_general_tests() {
+  PGUSER=user PASS=pass POSTGRESQL_MAX_CONNECTIONS=42 POSTGRESQL_MAX_PREPARED_TRANSACTIONS=42 POSTGRESQL_SHARED_BUFFERS=64MB run_tests no_admin
+  PGUSER=user1 PASS=pass1 ADMIN_PASS=r00t run_tests admin
+  DB=postgres ADMIN_PASS=r00t run_tests only_admin
+  # Test with arbitrary uid for the container
+  DOCKER_ARGS="-u 12345" PGUSER=user2 PASS=pass run_tests no_admin_altuid
+  DOCKER_ARGS="-u 12345" PGUSER=user3 PASS=pass1 ADMIN_PASS=r00t run_tests admin_altuid
+  DB=postgres DOCKER_ARGS="-u 12345" ADMIN_PASS=rOOt run_tests only_admin_altuid
+}
+
+function run_all_tests() {
+  for test_case in $TEST_LIST; do
+    : "Running test $test_case"
+    $test_case
+  done;
+}
+
 # configuration defaults
 POSTGRESQL_MAX_CONNECTIONS=100
 POSTGRESQL_MAX_PREPARED_TRANSACTIONS=0
 POSTGRESQL_SHARED_BUFFERS=32MB
 
-# Tests.
-
-run_container_creation_tests
-PGUSER=user PASS=pass POSTGRESQL_MAX_CONNECTIONS=42 POSTGRESQL_MAX_PREPARED_TRANSACTIONS=42 POSTGRESQL_SHARED_BUFFERS=64MB run_tests no_admin
-PGUSER=user1 PASS=pass1 ADMIN_PASS=r00t run_tests admin
-DB=postgres ADMIN_PASS=r00t run_tests only_admin
-# Test with arbitrary uid for the container
-DOCKER_ARGS="-u 12345" PGUSER=user2 PASS=pass run_tests no_admin_altuid
-DOCKER_ARGS="-u 12345" PGUSER=user3 PASS=pass1 ADMIN_PASS=r00t run_tests admin_altuid
-DB=postgres DOCKER_ARGS="-u 12345" ADMIN_PASS=rOOt run_tests only_admin_altuid
-run_change_password_test
-DB=postgres PGUSER=master PASS=master run_replication_test
-DB=postgres PGUSER=master PASS=master run_master_restart_test
+# Run all tests when the test case is not explicitly provided
+test $# -ge 1 || { run_all_tests && exit 0; }
+# Run the chosen test.
+: "Running test $1"
```

Comment:
I missed this `test $# -ge 1` before .. doesn't that mean that we should rather do:
```
for test in "$@"
do
   : "running test $test"
  "$test"
done
```